### DISCUSSION TITLE
Is it possible to round the bottom corners of windows?

### DISCUSSION THREAD

**Discussion by winteriscariot** [labwc/labwc#1324] [Q&A]
Mostly just curious, if it's a big deal then disregard, the current default is entirely workable. I'm just trying to replicate the feel of my old openbox+picom setup, in which picom rounds all 4 corners.

**Comment by johanmalm** [comment]
Not currently supported.
I don't think it would be that hard to code, given that we've already have rounded top corners. Conceptually, I'd be a bit unsure on approach... assume border thickness would have to be >= radius - which is generally thicker than most people would want it?

**Comment by Sinaaaa** [comment]
> Not currently supported. I don't think it would be that hard to code, given that we've already have rounded top corners. Conceptually, I'd be a bit unsure on approach... assume border thickness would have to be >= radius - which is generally thicker than most people would want it?

I think it would be great if it worked exactly the same as with the top corners. 

>border thickness would have to be >= radius

I'm not sure how that would work, but right now in my rc.xml radius is set at 8 & I have a 5 pixel wide border, the end result looks quite nice to my eyes.

**Reply by johanmalm** [reply]
I’m not how to explain the radius vs thickness thing, but I’ll try.
At the bottom, there is no titlebar server-side decoration to “eat into” - there is only the client surface and the border thickness. So if the radius is big, a bit of the client surface would stick out outside it. 
It is possible to clip the client surface, but I’m not convinced that’s a good thing to do, and I guess it would need to be a curved clip, the implementation of which is not obvious to me.

**Comment by Sinaaaa** [comment]
> I’m not how to explain the radius vs thickness thing, but I’ll try. At the bottom, there is no titlebar server-side decoration to “eat into” - there is only the client surface and the border thickness. So if the radius is big, a bit of the client surface would stick out outside it. It is possible to clip the client surface, but I’m not convinced that’s a good thing to do, and I guess it would need to be a curved clip, the implementation of which is not obvious to me.

Ah I understand now. Then yes what you said before makes complete sense to me. I think most people would be fine if only very small radii corners were possible. (unless they used very thick borders)

**Comment by ludgie** [comment]
Maybe round bottom corners would be feasible when handle and grip are supported
(https://github.com/labwc/labwc/discussions/1043)
